src/tsserver/server.ts

```diff
@@ -970,4 +970,12 @@ namespace ts.server {
     if (ts.sys.tryEnableSourceMapsForHost && /^development$/i.test(ts.sys.getEnvironmentVariable("NODE_ENV"))) {
         ts.sys.tryEnableSourceMapsForHost();
     }
+
+    // Overwrites the current console messages to instead write to
+    // the log. This is so that language service plugins which use
+    // console.log don't break the message passing between tsserver
+    // and the client
+    console.log = (msg) => logger.msg(msg, Msg.Info);

```

Comment:
console.log/warn/error are normally variadic in node, so would it make sense to do something with all `...args` in this replacement function?  We'd be less likely to lose important information if a plugin passed multiple things to one log call.

Reply:
I don't think the logger accepts variadic args - we can pass the array joined as a string if args.length is greater than 1 though.